keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -345,24 +345,31 @@ kind: Service
 metadata:
   name: my-service
 spec:
+  type: ClusterIP
+  ClusterIP: 1.2.3.4
   ipFamily: IPv6
   selector:
     app: MyApp
   ports:
     - protocol: TCP
       port: 80
       targetPort: 9376
+status:
+  preferredIPFamily:
+    - IPv46
 ```
 
 Dual-stack services (having both an IPv4 and IPv6 cluster IP) are currently out of scope of this proposal. Should there be use-cases where dual-stack Services are needed then we can revisit.
 
 #### Type Headless services
-
-\<TBD\>
+Headless services will operate using the following conditions:
+1. api-server does not default `ipFamily` field. User may optionally to choose to set it.
+2. If `ipFamily` field is set by the user, then the value of `preferredIPFamily` will follow accordingly. Example: setting `ipFamily` to `IPv6` then `preferredIPFamily` will have `IPv6` as a value.
+3. If `ipFamily` field is not set, then the value of `preferredIPFamily` will carry the values set in `-- service-cluster-ip-range`, which can be a single value in even in a dual stack cluster.
 
 #### Type ExternalName
 
-\<TBD\>
+Follows the same rules as `Type Headless Services`
```

Comment:
My understanding of `ExternalName` is that it generates a CNAME record only, so it's "transparent" to the whole IPFamily discussion. IIRC the debate on this type was about having a meaningless value or nil the field, but I'm not sure if that was just my interpretation 😅 

My suggestion for `Type ExternalName` is:

```
Services of type ExternalName map a Service to a DNS name. They create a CNAME record with the value on the field `externalName` mapped to the service DNS record.
CNAME record (Canonical Name Record) is used to maps an alias name to a true or canonical domain name. When a DNS resolver encounters a CNAME record, the DNS lookup will continue by retrying the lookup with true or canonical domain name.

IPFamily is not considered in these type of services and ...
```
* it should be nil 
* it should not be considered, and will adopt:
   * whatever value
   * preferredIPFamily
???